File: packages/ai-gateway/src/adapters/openai.ts
// Adapter OpenAI (API oficial, formato chat/completions).

import { OpenAiCompatibleProvider, keyFromEnv } from "./openai-compatible.js";

export class OpenAiProvider extends OpenAiCompatibleProvider {
  constructor() {
    super(
      "openai",
      ["gpt-4o-mini", "gpt-5"],
      {
        baseUrl: "https://api.openai.com/v1",
        label: "openai",
      },
      keyFromEnv("OPENAI_API_KEY"),
    );
  }
}

File: apps/api/internal/gateway/adapters/openai.go
package adapters

import "github.com/blu-ia/api/internal/gateway"

func NewOpenAIProvider() *OpenAICompatibleProvider {
	return NewOpenAICompatible(
		gateway.ProviderID("openai"),
		[]string{"gpt-4o-mini", "gpt-5"},
		OpenAICompatibleOptions{BaseURL: "https://api.openai.com/v1", Label: "openai"},
		KeyFromEnv("OPENAI_API_KEY"),
	)
}
